src/components/button/common/button-common.ts

```diff
@@ -0,0 +1,68 @@
+import type { CSSResultGroup, LitElement } from 'lit';
+import { property } from 'lit/decorators.js';
+
+import { type Constructor, NamedSlotStateController } from '../../core/common-behaviors';
+import { ACTION_ELEMENTS, hostContext, toggleDatasetEntry } from '../../core/dom';
+import { actionElementHandlerAspect, HandlerRepository } from '../../core/eventing';
+import type { IsStaticProperty } from '../../core/interfaces';
+
+import '../../icon';
+
+import style from './button.scss?lit&inline';
+
+export type SbbButtonSize = 'l' | 'm';
+export type SbbButtonVariant = 'primary' | 'secondary' | 'tertiary' | 'transparent';
+
+export declare class SbbButtonCommonInterface {
+  public variant: SbbButtonVariant;
+  public size?: SbbButtonSize;
+  public isStatic: boolean;
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const SbbButtonCommonElementMixin = <T extends Constructor<LitElement>>(
+  superClass: T,
+): Constructor<SbbButtonCommonInterface> & T => {
+  class SbbButtonCommonElement
+    extends superClass
+    implements Partial<SbbButtonCommonInterface>, IsStaticProperty
+  {
+    public static styles: CSSResultGroup = style;
+    /** Variant of the button, like primary, secondary etc. */
+    @property({ reflect: true }) public variant: SbbButtonVariant = 'primary';
+
+    /** Size variant, either l or m. */
+    @property({ reflect: true }) public size?: SbbButtonSize = 'l';
+
+    /**
+     * Set this property to true if you want only a visual representation of a
+     * button, but no interaction (a span instead of a link/button will be rendered).
+     */
+    @property({ attribute: 'is-static', reflect: true, type: Boolean }) public isStatic = false;
+
+    private _handlerRepository = new HandlerRepository(this, actionElementHandlerAspect);
```

Comment:
No, I'll start working on it tomorrow, together with https://github.com/lyne-design-system/lyne-components/pull/2364#discussion_r1470708971

Reply:
@kyubisation i have a question about this:
right now, the static button emits the clicked event even if disabled; i feel it's wrong and i should prevent it, right?

Reply:
feels wrong to me too, I think it should not dispatch any event

Reply:
Well, it's complicated. 😅 
In the static variant, it has no true understanding of disabled, because it is not an interactive element. Therefore it should allow a click event to emit, even if it is disabled, because the parent interactive element should suppress the click.
https://stackblitz.com/edit/js-qlz5c9?file=index.js